app/scripts/controllers/modals/logoutModal.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+/* jshint unused: false */
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:LogoutModalController
+ * @description
+ * # LogoutModalController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('LogoutModalController', function ($location, $filter, $scope, $uibModalInstance, Constants) {
+    $scope.endTimestamp = moment().add(30, 'seconds');
+    var lastInteraction = Date.parse(localStorage.getItem('origin-web-console-last-interaction-timestamp'));
+    var countdownInterval = setInterval(function() {
+      var timeRemaining = $filter('countdownToTimestamp')($scope.endTimestamp);
+      $('.countdown').text(function() {
+        return timeRemaining;
+      });
+      if (timeRemaining == 0) {
+        $scope.logout();
+      }
+    }, 1000);
```

Comment:
instead of an interval here you just need a single timeout of 30 seconds that runs the logout, clear the timeout below instead of clear interval

Reply:
But the interval makes sure that the displayed countdown number will be count down, the same as in https://github.com/openshift/origin-web-console/blob/master/app/scripts/app.js#L491-L505